customArguments.go

```diff
@@ -36,47 +36,87 @@ func newCustomArguments(stub *mockFunction, arguments []interface{}) *customArgu
 	}
 }
 
+// isArgumentLengthValid - returns whether or not the length of the provided arguments
+// are valid for the stub. Taking in variadic function into account.
+func isArgumentLengthValid(functionType reflect.Type, arguments []interface{}) bool {
+	if functionType.IsVariadic() {
+		return len(arguments) < functionType.NumIn()-1
+	}
+
+	return len(arguments) != functionType.NumIn()
+}
+
 // getMatchers returns a slice of matchers based on the types and values of the provided arguments
 func getMatchers(functionType reflect.Type, arguments []interface{}) ([]match.SupportedKindsMatcher, error) {
 	matchers := make([]match.SupportedKindsMatcher, functionType.NumIn())
-	for i, arg := range arguments {
+	for i := 0; i < functionType.NumIn(); i++ {
 		aType := functionType.In(i)
 
-		switch arg.(type) {
-		case match.SupportedKindsMatcher:
-			matcher := arg.(match.SupportedKindsMatcher)
-			if _, ok := matcher.SupportedKinds()[aType.Kind()]; !ok {
-				return nil, &argumentValidationError{
-					fnType:   functionType,
-					provided: arguments,
-				}
+		if isVariadicArgument(i, functionType) {
```

Comment:
I think I should swap the argument order here to keep the `functionType` as the first argument.